### PR TITLE
imp: remove `LatestSequenceSend`

### PR DESCRIPTION
**Pull request by sangier**
## Description

closes: [#5005](https://github.com/cosmos/ibc-go/issues/5005)

### Commit Message / Changelog Entry

```text
del: latestsequencesend field
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by crodriguezvega — APPROVED** [review]
Thank you, @sangier! I just left comment for a test case that I think we can keep.

File: modules/core/04-channel/keeper/packet_test.go [hunk 1]
```diff
@@ -335,82 +335,6 @@ func (suite *KeeperTestSuite) TestRecvPacket() {
...
-				packet = types.NewPacket(ibctesting.MockPacketData, sequence, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
-				channelCap = suite.chainB.GetChannelCapability(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
-
-				channel := path.EndpointB.GetChannel()
-				channel.State = types.FLUSHING
-				path.EndpointB.SetChannel(channel)
-
-				// set last packet sent sequence to sequence + 1
-				counterpartyUpgrade := types.Upgrade{LatestSequenceSend: sequence + 1}
-				path.EndpointB.SetChannelCounterpartyUpgrade(counterpartyUpgrade)
```
[the remark this answers is not in the document]
Comment (sangier) [reply modules/core/04-channel/keeper/packet_test.go:353]: Re-added the test removing the highlighted lines!

File: e2e/testsuite/grpc_query.go [hunk 2]
```diff
@@ -85,15 +85,15 @@ func (s *E2ETestSuite) InitGRPCClients(chain *cosmos.CosmosChain) {
...
-		UpgradeQueryClient:     upgradetypes.NewQueryClient(grpcConn),
+		BankQueryClient:          banktypes.NewQueryClient(grpcConn),
+		GovQueryClient:           govtypesv1beta1.NewQueryClient(grpcConn),
+		GovQueryClientV1:         govtypesv1.NewQueryClient(grpcConn),
+		GroupsQueryClient:        grouptypes.NewQueryClient(grpcConn),
+		ParamsQueryClient:        paramsproposaltypes.NewQueryClient(grpcConn),
+		AuthQueryClient:          authtypes.NewQueryClient(grpcConn),
+		AuthZQueryClient:         authz.NewQueryClient(grpcConn),
+		ConsensusServiceClient:   cmtservice.NewServiceClient(grpcConn),
+		UpgradeQueryClient:       upgradetypes.NewQueryClient(grpcConn),
```
Comment (chatton) [inline e2e/testsuite/grpc_query.go:96]: seems like something weird is going on with linting 🤔

**Review by chatton — APPROVED** [review]
LGTM thanks @sangier!

File: e2e/testsuite/grpc_query.go (hunk 2 above)
Comment (crodriguezvega) [reply e2e/testsuite/grpc_query.go:96]: I pushed this change after I run gofumpt in the branch.

**Pull request merged**